### PR TITLE
Runtime Performance Optimization

### PR DESCRIPTION
**Pull request by kateinoigakukun**
- Reduce unnecessary calls of `memory` and `dataView` getter in runtime TS code
- Promote result pointer to return value to reduce memory operations
- Reduce Swift ARC retain/release operations by:
  - Reduce Optional wrapper
  - Reduce reference escapes
- Add fast paths for empty arguments call

### CODE REVIEW

File: Runtime/src/index.ts [hunk 1]
```diff
@@ -309,37 +284,21 @@ export class SwiftRuntime {
...
-            kind_ptr: pointer,
             payload1_ptr: pointer,
             payload2_ptr: pointer
         ) => {
             const obj = this.memory.getObject(obj_ref);
             const func = this.memory.getObject(func_ref);
-            let isException = true;
-            try {
+            let result = undefined;
                 const args = JSValue.decodeArray(argv, argc, this.memory);
```
Comment (sunnypp) [inline Runtime/src/index.ts:293]: minor: indentation

File: Runtime/src/index.ts [hunk 2]
```diff
@@ -127,28 +121,28 @@ export class SwiftRuntime {
...
             payload2_ptr: pointer
         ) => {
-            const obj = this.memory.getObject(ref);
-            const key = this.memory.getObject(name);
+            const memory = this.memory;
+            const obj = memory.getObject(ref);
+            const key = memory.getObject(name);
             const result = obj[key];
-            JSValue.write(
+            return JSValue.writeV2(
```
Comment (MaxDesiatov) [inline Runtime/src/index.ts:140]: Is the old version of `write` still used anywhere? Should it be renamed to `writeV1` then?

Comment (kateinoigakukun) [reply Runtime/src/index.ts]: The old one is still used for writing out multiple values (e.g. passing js values to Swift closure )

File: Runtime/src/index.ts [hunk 3]
```diff
@@ -127,28 +121,28 @@ export class SwiftRuntime {
...
-            const obj = this.memory.getObject(ref);
-            const key = this.memory.getObject(name);
-            const value = JSValue.decode(kind, payload1, payload2, this.memory);
+            const memory = this.memory;
+            const obj = memory.getObject(ref);
+            const key = memory.getObject(name);
+            const value = JSValue.decode(kind, payload1, payload2, memory);
             obj[key] = value;
         },
         swjs_get_prop: (
```
Comment (MaxDesiatov) [inline Runtime/src/index.ts:130]: Would you mind adding doc comments to functions in `wasmImports`, including their parameters and return values/types? Thanks!

File: Sources/_CJavaScriptKit/include/_CJavaScriptKit.h [hunk 4]
```diff
@@ -175,66 +178,72 @@ extern JavaScriptObjectRef _i64_to_bigint_slow(unsigned int lower, unsigned int
...
 
 /// `_call_function` calls JavaScript function with given arguments list without capturing any exception
 ///
 /// @param ref The target JavaScript function to call.
 /// @param argv A list of `RawJSValue` arguments to apply.
 /// @param argc The length of `argv``.
-/// @param result_kind A result pointer of JavaScript value kind of returned result or thrown exception.
 /// @param result_payload1 A result pointer of first payload of JavaScript value of returned result or thrown exception.
 /// @param result_payload2 A result pointer of second payload of JavaScript value of returned result or thrown exception.
+/// @return A JavaScriptValueKindAndFlags bits represented as 32bit integer for the returned value.
```
Comment (MaxDesiatov) [inline Sources/_CJavaScriptKit/include/_CJavaScriptKit.h:200]:
```suggestion
/// @return A `JavaScriptValueKindAndFlags` bits represented as 32bit integer for the returned value.
```

**Review by MaxDesiatov — APPROVED** [review]
Fantastic work, these are very impressive improvements! 👏

I only have a couple of suggestions for improving doc comments, but otherwise this LGTM.

File: Runtime/src/index.ts (hunk 3 above)
Comment (kateinoigakukun) [reply Runtime/src/index.ts:130]: They would be exactly same as `_CJavaScriptKit.h` one. Do you mean we need to add docs here as well for library users?

Comment (MaxDesiatov) [reply Runtime/src/index.ts:130]: Ah, makes sense. Then we probably don't need those duplicated I think. Thanks for checking!

**Review by MaxDesiatov — APPROVED** [review]

**Pull request merged**